spec/models/gestionnaire_spec.rb

```diff
@@ -78,4 +79,105 @@
+        expect(gestionnaire.unread_commentaires?(groupe_gestionnaire)).to eq false
+        expect(gestionnaire_unread_commentaire_cause_never_seen.unread_commentaires?(groupe_gestionnaire_unread_commentaire_cause_never_seen)).to eq true
+        expect(gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire.unread_commentaires?(groupe_gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire)).to eq true
+      end
+    end
+
+    context "over same groupe_gestionnaire" do
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:gestionnaire_unread_commentaire_cause_never_seen) { create(:gestionnaire) }
+      let(:gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire, gestionnaire_unread_commentaire_cause_never_seen, gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+
+      let!(:follow_commentaire_groupe_gestionnaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire, sender: administrateur, commentaire_seen_at: Time.zone.now) }
+      let!(:follow_commentaire_groupe_gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire, sender: administrateur, commentaire_seen_at: 1.day.ago) }
+
+      it do
+        expect(gestionnaire.unread_commentaires?(groupe_gestionnaire)).to eq false
+        expect(gestionnaire_unread_commentaire_cause_never_seen.unread_commentaires?(groupe_gestionnaire)).to eq true
+        expect(gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire.unread_commentaires?(groupe_gestionnaire)).to eq true
+      end
+    end
+  end
+
+  describe "#commentaire_seen_at" do
+    context "when already seen commentaire" do
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+      let!(:follow_commentaire_groupe_gestionnaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire, sender: administrateur, commentaire_seen_at: Time.zone.now) }
+
+      it { expect(gestionnaire.commentaire_seen_at(groupe_gestionnaire, administrateur)).not_to eq nil }
```

Comment:
tant qu'à faire, autant s'assurer qu'on renvoie bien un datetime (modulo les microsecondes)
```suggestion
      it { expect(gestionnaire.commentaire_seen_at(groupe_gestionnaire, administrateur.to_date)).to eq Date.current }
```

Reply:
pas sur de comprendre pourquoi
`administrateur.to_date`

Reply:
done